### ISSUE TITLE
Fix Nightly Build (upgrade hashbrown)

### ISSUE THREAD

**Issue by KCarretto** [facebookexperimental/starlark-rust#108]
# Problem Overview

* `hashbrown` v0.12.3 (currently locked to) depends on `ahash` v0.7.7
* `ahash` v0.7.7 no longer builds on `nightly-2024-02-06` due to the feature `stdsimd` being removed (see [this issue](https://github.com/tkaitchuck/aHash/issues/200))
```
error[E0635]: unknown feature `stdsimd`
  --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/ahash-0.7.7/src/lib.rs:33:42
   |
33 | #![cfg_attr(feature = "stdsimd", feature(stdsimd))]
   |  
```
* `hashbrown` v0.14.3 resolves this issue ([by upgrading to `ahash` v0.8.6](https://github.com/rust-lang/hashbrown/blob/f2e62124cd947b5e2309dd6a24c7e422932aae97/Cargo.toml#L17))

# Requested Solution

* Upgrade `hashbrown` dependency to `v0.14.3`

**Comment by wendy728** [comment]
@dtolnay is already working on the upgrade

**Comment by dtolnay** [comment]
Fixed in `main`. @wendy728 would you take care of publishing a release to crates.io? The most recent `starlark` release still has a dependency on hashbrown 0.12.3 so it's still broken.

**Comment by wendy728** [comment]
Released [starlark-rust v0.12.0](https://github.com/facebookexperimental/starlark-rust/releases/tag/v0.12.0)
